microprofile/reactive-streams/src/main/java/io/helidon/microprofile/reactive/MultiStagesCollector.java

```diff
@@ -0,0 +1,231 @@
+ */
+class MultiStagesCollector<T> implements Collector<Stage, Multi<T>, CompletionStage<T>> {
+
+    private Multi<T> multi = null;
+    private List<Flow.Processor<Object, Object>> processorList = new ArrayList<>();
+    private CompletionStage<Object> completionStage = null;
+    private HelidonSubscriberWithCompletionStage<T> subscriberWithCompletionStage = null;
+
+    @Override
+    public Supplier<Multi<T>> supplier() {
+        return () -> multi != null ? multi : Multi.empty();
+    }
+
+    @SuppressWarnings("unchecked")
+    private void subscribeUpStream() {
+        if (multi != null) {
+            for (Flow.Processor p : processorList) {
+                multi.subscribe(p);
+                multi = (Multi<T>) p;
+            }
+        } else {
+            throw new RuntimeException("No producer was supplied");
+        }
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public BiConsumer<Multi<T>, Stage> accumulator() {
+        //MP Stages to Helidon multi streams mapping
+        return (m, stage) -> {
+
+            if (stage instanceof Stage.PublisherStage) {
+                Stage.PublisherStage publisherStage = (Stage.PublisherStage) stage;
+                Publisher<T> rsPublisher = (Publisher<T>) publisherStage.getRsPublisher();
+                multi = MultiRS.toMulti(rsPublisher);
+
+            } else if (stage instanceof Stage.Of) {
+                Stage.Of stageOf = (Stage.Of) stage;
+                List<?> fixedData = StreamSupport.stream(stageOf.getElements().spliterator(), false)
+                        .collect(Collectors.toList());
+                multi = (Multi<T>) Multi.just(fixedData);
+
+            } else if (stage instanceof Stage.Map) {
+                Stage.Map mapStage = (Stage.Map) stage;
+                Function<Object, Object> mapper = (Function<Object, Object>) mapStage.getMapper();
+                processorList.add(new MultiMappingProcessor<>(mapper::apply));
+
+            } else if (stage instanceof Stage.Filter) {
+                Stage.Filter stageFilter = (Stage.Filter) stage;
+                Predicate<T> predicate = (Predicate<T>) stageFilter.getPredicate();
+                processorList.add(new FilterProcessor(predicate));
+
+            } else if (stage instanceof Stage.Peek) {
+                Stage.Peek peekStage = (Stage.Peek) stage;
+                Consumer<Object> peekConsumer = (Consumer<Object>) peekStage.getConsumer();
+                processorList.add(new PeekProcessor<Object>(peekConsumer));
+
+            } else if (stage instanceof Stage.Limit) {
+                Stage.Limit limitStage = (Stage.Limit) stage;
+                processorList.add(new LimitProcessor(limitStage.getLimit()));
+
+            } else if (stage instanceof Stage.FlatMap) {
+                Stage.FlatMap flatMapStage = (Stage.FlatMap) stage;
+                Function<?, Graph> mapper = flatMapStage.getMapper();
+                processorList.add(new FlatMapProcessor(mapper));
```

Comment:
@romain-grecourt I am afraid I need Multi processors for backing the MicroProfile Reactive Streams Operators, only way to keep them package protected would be reusing package `io.helidon.common.reactive` in different project, namely `microprofile/reactive-streams` which is possible, it just doesn't seems right

Reply:
There is also a possibility to move MP Reactive Streams operators impl to `common-reactive` but its MP api it seems appropriate to place it among others MP modules

Reply:
OK, I searched for usages of it in this diff when I wrote my comment but didn't find any. If you need that to be used in other modules the only way is to make it public. Modules must have different packages.